src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3010,13 +3046,113 @@ export class StudyViewPageStore
     ): void {
         trackStudyViewFilterEvent('genomicCategoricalData', this);
 
+        const chart = this._geneSpecificChartMap.get(uniqueKey);
         const dataFilterValues: DataFilterValue[] = getCategoricalFilterValues(
             values
         );
-        this.updateGenomicDataFiltersByValues(
-            uniqueKey,
-            _.cloneDeep(dataFilterValues)
+        if (chart!.mutationOptionType) {
+            this.updateMutationDataFilters(uniqueKey, [_.cloneDeep(values)]);
```

Comment:
i'm curious why this needs to be cloned.  did you take that from elsewhere?

Reply:
I tested it before where the variable values somehow get dependent on the original place. So I want it to be clean and new whenever possible.